src/main/scala/nl.knaw.dans.easy.curationwork/Unassign.scala

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.curationwork
+
+import java.nio.file.{Files, Path}
+
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.apache.commons.io.FileUtils
+import resource.managed
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.io.StdIn
+import scala.language.postfixOps
+import scala.sys.process._
+import scala.util.Try
+
+class Unassign(commonCurationDir: Path, managerCurationDirString: String) extends EasyManageCurationWorkApp(commonCurationDir, managerCurationDirString) with DebugEnhancedLogging {
+
+  private def getCurrentUnixUser: String = {
+    "whoami" !!
+  }
+
+  private def isSubmitted(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getString("state.label") == "SUBMITTED"
+  }
+
+  private def isCurated(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getString("curation.performed") == "yes"
+  }
+
+  private def clearProperties(depositProperties: PropertiesConfiguration): Unit = {
+    depositProperties.clearProperty("curation.datamanager.userId")
+    depositProperties.clearProperty("curation.datamanager.email")
+    depositProperties.save()
+  }
+
+  @tailrec
+  private def confirmAction(datamanager: DatamanagerId): Boolean = {
+    StdIn.readLine(s"This action will move all deposits from the curation area of datamanager $datamanager to the common curation area. OK? (y/n):") match {
+      case "y" => true
+      case "n" => false
+      case _ =>
+        println("Please enter a valid char : y or n ")
+        confirmAction(datamanager)
+    }
+  }
+
+  private def unassignDeposit(deposit: Path, datamanager: DatamanagerId): String = {
+    val depositProperties = new PropertiesConfiguration(deposit.resolve("deposit.properties").toFile)
+    if (!isSubmitted(depositProperties) || isCurated(depositProperties)) {
+      var msg = ""
+      if (!isSubmitted(depositProperties))
+        msg = s"\nError: Deposit $deposit is not SUBMITTED. It was not unassigned."
+      if (isCurated(depositProperties))
+        msg += s"\nError: Deposit $deposit has already been curated. It was not unassigned."
+      msg
+    }
+    else {
+      clearProperties(depositProperties)
+      FileUtils.moveDirectory(deposit.toFile, commonCurationDir.resolve(deposit.getFileName).toFile)
+      s"\nDeposit $deposit has been unassigned from datamanager $datamanager."
+    }
+  }
+
+  private def unassignFromDatamanager(personalCurationDirectory: Path, uuid: Option[Uuid], datamanager: DatamanagerId): String = {
+    uuid match {
+      case Some(deposit) => unassignDeposit(personalCurationDirectory.resolve(deposit), datamanager)
+      case None =>
+        val msg = managed(Files.list(personalCurationDirectory)).acquireAndGet(jStream => jStream.iterator().asScala.toList)
```

Comment:
I have been hesitating to suggest the better-files library for this project until now, given that most of the things you do with files are `Files.exists`, `FileUtils.move`, `Path.resolve`, etc. But now that I see that you're using `Files.list` here (I may or may not have had something to do with the `managed` that is wrapping this stuff, I vaguely remember), I think it is time for me to officially suggest to you the use of [better-files](https://github.com/pathikrit/better-files). With that all these file operations become much simpler, including this one. The result for this project will probably a total replacement of `java.nio.file.Path`/`Paths`/`Files` and total removal of the `apache/commons-io` package.

Reply:
better-files in use.  It is worth its name!